### PR TITLE
ref(js): Move context icons to a codesplit react component

### PR DESCRIPTION
**Pull request by evanpurkhiser**
Deletes a bunch of CSS and moves all the icons into a codesplit file.

This cuts down on the sentry.css file by ~ 40%

### CODE REVIEW

**Review by dashed — APPROVED** [review]

File: static/app/components/events/contextSummary/filterContexts.tsx [hunk 1]
```diff
@@ -1,29 +0,0 @@
...
-      !(event.contexts.runtime?.name || event.contexts.runtime?.version)
-    ) {
-      return false;
-    }
-  }
-
-  return true;
-}
-
-export default filterContexts;
```
Comment (evanpurkhiser) [inline static/app/components/events/contextSummary/filterContexts.tsx:29]: Moved to `utils`

File: static/app/components/events/contextSummary/generateClassName.tsx [hunk 2]
```diff
@@ -1,46 +0,0 @@
...
-
-    if (isLegacyEdge) {
-      return 'legacy-edge';
-    }
-  }
-
-  return formattedName;
-}
-
-export default generateClassname;
```
Comment (evanpurkhiser) [inline static/app/components/events/contextSummary/generateClassName.tsx:46]: Moved to `utils`

File: static/app/components/events/contextSummary/index.tsx [hunk 3]
```diff
@@ -35,85 +32,93 @@ const KNOWN_CONTEXTS: Context[] = [
...
+    Component: ContextSummaryOS,
+  },
+  {
+    keys: ['device'],
+    Component: ContextSummaryDevice,
+  },
+  {
+    keys: ['gpu'],
+    Component: ContextSummaryGPU,
+  },
```
Comment (evanpurkhiser) [inline static/app/components/events/contextSummary/index.tsx:46]: Formatting changes for consistency

File: static/app/components/events/contextSummary/index.tsx [hunk 4]
```diff
@@ -35,85 +32,93 @@ const KNOWN_CONTEXTS: Context[] = [
...
-  // Add defined contexts in the declared order, until we reach the limit
-  // defined by MAX_CONTEXTS.
-  let contexts = KNOWN_CONTEXTS.filter(context => filterContexts(event, context)).map(
-    ({keys, Component, unknownTitle, omitUnknownVersion}) => {
-      if (contextCount >= MAX_CONTEXTS) {
-        return null;
-      }
+  // XXX: We want to have *at least* MIN_CONTEXTS, so we first find all the
+  // contexts that have data, if that doesn't complete our MIN_CONTEXTS, we add
+  // in contextrs that have no data.
```
Comment (evanpurkhiser) [inline static/app/components/events/contextSummary/index.tsx:58]: I rewrote this to be a lot more clear, it's explained here what it's trying to do.

File: static/app/components/events/contextSummary/item.tsx [hunk 5]
```diff
@@ -26,20 +44,42 @@ const Details = styled('div')`
...
+    margin-bottom: ${space(0.25)};
+  }
+
+  p {
+    font-size: ${p => p.theme.fontSizeSmall};
+
+    &:last-child {
+      margin: 0;
+    }
+  }
```
Comment (evanpurkhiser) [inline static/app/components/events/contextSummary/item.tsx:74]: This more or less came directly out of the less file

File: static/app/components/events/contextSummary/types.tsx [hunk 6]
```diff
@@ -0,0 +1,23 @@
+import {Event} from 'sentry/types';
+
+type Meta = NonNullable<Event['_meta']>;
```
Comment (evanpurkhiser) [inline static/app/components/events/contextSummary/types.tsx:3]: This is already `any`, but maybe in the future will be better

File: static/app/components/events/eventTagsAndScreenshot/eventTagsAndScreenshot.spec.tsx [hunk 7]
```diff
@@ -438,7 +438,7 @@ describe('EventTagsAndScreenshot', function () {
       // Screenshot Container
       expect(screen.getByText('Screenshot')).toBeInTheDocument();
       expect(screen.getByText('View screenshot')).toBeInTheDocument();
-      expect(screen.getByRole('img')).toHaveAttribute(
+      expect(screen.getByTestId('image-viewer')).toHaveAttribute(
```
Comment (evanpurkhiser) [inline static/app/components/events/eventTagsAndScreenshot/eventTagsAndScreenshot.spec.tsx:441]: Had to change these since there's more than one `img` now

**Pull request merged**